lib/router/directory-router.js

```diff
@@ -297,6 +297,9 @@ function DirectoryRouter(root, runas, dirIndex, prefix, logdir, settings,
                 var rmd = _.find(entries, function(entry) {
                   return /\.rmd$/i.test(entry);
                 });
+                var uiRmd = _.find(entries, function(entry) {
```

Comment:
The name `ui.Rmd` is unimportant; the filename is a convention, not a requirement. See: https://github.com/rstudio/rsconnect/pull/461#discussion_r506707936

This was my original approach in rsconnect, too, before looking more deeply at each Rmd to understand if it asks for a Shiny runtime.

Reply:
Crap, I was hoping to avoid doing that, oh well. Thanks for catching that.